src/test/java/com/rybka/service/export/JsonExportServiceTest.java

```diff
@@ -0,0 +1,122 @@
+package com.rybka.service.export;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rybka.configuration.ExportProperty;
+import com.rybka.exception.ExportFailureException;
+import com.rybka.model.CurrencyHistory;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Random;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+class JsonExportServiceTest {
+
+    @Mock
+    private ExportProperty exportProperty;
+    @Mock
+    private ObjectMapper mapper;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final Random random = new Random();
+    private JsonExportService jsonExportService;
+
+    @Test
+    public void testOnProperExport() {
+
+        // given
+        var resourceFolder = "src/test/resources/";
+        var tempFolderPath = Paths.get(resourceFolder + random.nextInt() + "/");
+        var tempFolder = createTestDirectory(tempFolderPath);
+        var exportType = "json";
+        var history = buildRandomHistory();
+        var testList = List.of(history);
+        jsonExportService = new JsonExportService(objectMapper, exportProperty);
+
+        when(exportProperty.getFolder()).thenReturn(tempFolder.toString() + "/");
+        when(exportProperty.getType()).thenReturn(exportType);
+
+        // when
+        jsonExportService.export(testList);
+        Path testFilePath = findSavedFile(tempFolder);
+        var actualResult = readSavedFile(testFilePath);
+
+        // then
+        verify(exportProperty).getType();
+        verify(exportProperty).getFolder();
+
+        Assertions.assertEquals(history, actualResult);
+
+        removeSavedFile(testFilePath);
+        removeSavedFile(tempFolderPath);
+    }
+
+    @SneakyThrows
+    @Test
+    public void testOnExportException() {
+
+        // given
+        var resourceFolder = "src/test/resources/";
+        var exportType = "json";
+        var testList = List.of(buildRandomHistory());
+        jsonExportService = new JsonExportService(mapper, exportProperty);
+
+        when(exportProperty.getType()).thenReturn(exportType);
+        when(exportProperty.getFolder()).thenReturn(resourceFolder);
+
+        // when
+        when(mapper.writeValueAsString(any(List.class))).thenThrow(ExportFailureException.class);
```

Comment:
Not sure that writeValueAsString throws ExportFailureException kind of exception.

Reply:
fixed